zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationMigrateProcessor.java

```diff
@@ -307,6 +318,31 @@ private void tryMigrateElementInstance(
             .setVersion(targetProcessDefinition.getVersion())
             .setElementId(targetElementId));
 
+    final int numberOfTakenSequenceFlows =
+        elementInstanceState.getNumberOfTakenSequenceFlows(elementInstance.getKey());
+    if (numberOfTakenSequenceFlows > 0) {
+      final Set<ExecutableSequenceFlow> sequenceFlows =
+          getSequenceFlowsToMigrate(
+              sourceProcessDefinition,
+              targetProcessDefinition,
+              sourceElementIdToTargetElementId,
+              elementInstance);
+
+      sequenceFlows.forEach(
+          sequenceFlow -> {
+            final var sequenceFlowRecord = new ProcessInstanceRecord();
+            sequenceFlowRecord.copyFrom(elementInstanceRecord);
+            sequenceFlowRecord
+                .setElementId(sequenceFlow.getId())
+                .setBpmnElementType(sequenceFlow.getElementType())
+                .setBpmnEventType(sequenceFlow.getEventType())
+                .setFlowScopeKey(elementInstance.getKey());
```

Comment:
💭 There may be a need to adjust the tree paths on migration, but we also don't take care of this for the active element instance yet.

🔧 We should probably reset the tree paths until we implement something correctly for it, to avoid filling the record with incorrect data

Reply:
@korthout I am a bit off from the tree path topic. Could you please clarify what needs to be done here?

Reply:
For now let's add this:
```suggestion
            sequenceFlowRecord
                .setElementId(sequenceFlow.getId())
                .setBpmnElementType(sequenceFlow.getElementType())
                .setBpmnEventType(sequenceFlow.getEventType())
                .setFlowScopeKey(elementInstance.getKey())
                .resetElementInstancePath()
                .resetCallingElementPath()
                .resetProcessDefinitionPath();
```